templates/terraform/custom_expand/subnetwork_log_config.go.erb

```diff
@@ -35,6 +35,14 @@ func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d T
 	transformed["aggregationInterval"] = original["aggregation_interval"]
 	transformed["flowSampling"] = original["flow_sampling"]
 	transformed["metadata"] = original["metadata"]
+	
+	// make it JSON marshallable
+	transformed["metadataFields"] = original["metadata_fields"].(*schema.Set).List()
+
+	// allow a null default (must eval to true otherwise)
+	if val := reflect.ValueOf(original["filter_expr"]); val.IsValid() && !isEmptyValue(val) {
```

Comment:
I'm  not 100% sure, but I think this is going to mean that `filter_expr` will not be able to be updated to an empty value after its been set (it depends on how the API handles unset fields in the PATCH request). Can you check?

Reply:
I've added a test that I hope covers unsetting this and the metadata fields. The API did not accept an empty value, but would accept a unset value.

Reply:
Oops.. test is failing. I'll take a closer look.

Reply:
I'm having some trouble with this. 

If it's unset, I think it is not updating.

If it sends a empty string it gets this error:

```
Error: Error creating Subnetwork: googleapi: Error 400: Invalid value for field 'resource.logConfig.filterExpr': ''. ERROR: VPC Flow Log filter expression:1:0: mismatched input '<EOF>' expecting {'[', '{', '(', '.', '-', '!', 'true', 'false', 'null', NUM_FLOAT, NUM_INT, NUM_UINT, STRING, BYTES, IDENTIFIER}
    | 
    | ^
ERROR: VPC Flow Log filter expression:1:0: Filter expression does not evaluate to a boolean
    | 
    | ^, invalid

    on /tmp/tf-test941063018/main.tf line 7:
    (source code not available)
```

Is the solution to send the filter expression string `true` when it's unset? That feels wrong.

Reply:
That does feel wrong. If you have the exact request/response from GCP (which you can get by running the test with env var TF_LOG=DEBUG) I can help look into it.